framework/graphics/dx12_resource_data_util.cpp

```diff
@@ -688,6 +688,7 @@ Dx12ResourceDataUtil::ExecuteCopyCommandList(ID3D12Resource*
 
                 // Prepare resource state.
                 AddTransitionBarrier(command_list_, target_resource, i, before_states[i], copy_state);
+                dx12::ResourceStateInfo promoted_state = copy_state;
```

Comment:
Would it be better to rename:

`copy_state` to `pre_copy_state` 
`promoted_state` to `post_copy_state` 

?